lib/Result.cpp

```diff
@@ -0,0 +1,61 @@
+
+#include "Result.h"
+
+#include "llvm/IR/DebugInfo.h"
+#include "llvm/IR/DebugInfoMetadata.h"
+#include "llvm/IR/Function.h"
+
+#include <sstream>
+#include <string>
+
+using namespace Mutang;
+using namespace llvm;
+
+std::vector<std::string> Result::calculateCallerPath(MutationResult *mutationResult) {
+  Testee *currentTestee = mutationResult->getTestee();
+
+  std::vector<std::string> callerPath;
+
+  /// Currently we don't print functions, only "file:line". Later we will and
+  /// for that we will have to demangle them.
+
+  /// Last path component: mutation point itself.
+  auto mutationPoint = mutationResult->getMutationPoint();
+  Instruction *instruction = dyn_cast<Instruction>(mutationPoint->getOriginalValue());
+  std::string fileName = instruction->getDebugLoc()->getFilename();
+  std::string line = std::to_string(instruction->getDebugLoc()->getLine());
+
+  ///Function *function = instruction->getFunction();
+  ///assert(function && "Expected function");
+  ///std::string functionName = function->getName().str();
+
+  std::stringstream mpComponentAsStream;
+  mpComponentAsStream << fileName.c_str();
+  mpComponentAsStream << ":";
+  mpComponentAsStream << line.c_str();
+  std::string mpComponent = mpComponentAsStream.str();
```

Comment:
You're not reusing this string so you can save an allocation by doing `callerPath.push_back(mpComponentAsStream.str())` directly.